aws/ec2/example.tf

```diff
@@ -0,0 +1,120 @@
+resource "aws_iam_role" "secrethub_demo" {
+  name               = "SecretHubDemoEC2Role"
+  assume_role_policy = data.aws_iam_policy_document.ec2_assume_role.json
+  description        = "Role for SecretHub demo app"
+}
+
+data "aws_iam_policy_document" "ec2_assume_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_kms_key" "secrethub_auth" {
+  description = "KMS key to facilitate SecretHub authentication"
+}
+
+data "aws_iam_policy_document" "secrethub_auth" {
+  statement {
+    actions   = ["kms:Decrypt"]
+    resources = [aws_kms_key.secrethub_auth.arn]
+    effect    = "Allow"
+  }
+}
+
+resource "aws_iam_policy" "secrethub_auth" {
+  name        = "SecretHubAuth"
+  description = "Allow SecretHub authentication using KMS"
+  policy      = data.aws_iam_policy_document.secrethub_auth.json
+}
+
+resource "aws_iam_role_policy_attachment" "secrethub_demo_auth" {
+  role       = aws_iam_role.secrethub_demo.name
+  policy_arn = aws_iam_policy.secrethub_auth.arn
+}
+
+variable "secrethub_repo" {
+  description = "The name of your SecretHub repository."
+}
```

Comment:
In the GCP examples we're now using the `variables.tf` and `outputs.tf` convention, we could do that here too.